src/ess/reflectometry/conversions.py

```diff
@@ -126,126 +117,39 @@ def specular_reflection(
-
-    Parameters
-    ----------
-    data_array:
-        Data array to convert.
-    graph:
-        Graph for :code:`transform_coords`.
-    wavelength_edges:
-        Bounds for wavelength values, exclude data outside of bounds.
-
-    Returns
-    -------
-    :
-        New data array with wavelength dimension.
-    """
-    data_array_wav = data_array.transform_coords(["wavelength"], graph=graph)
-    if wavelength_edges is not None:
-        data_array_wav = data_array_wav.bin({wavelength_edges.dim: wavelength_edges})
-    return WavelengthData[Run](data_array_wav)
-
-
-def wavelength_to_theta(
-    data_array: WavelengthData[Run],
+def add_coords(
+    da: ChopperCorrectedTofEvents[Run],
     graph: SpecularReflectionCoordTransformGraph[Run],
-) -> ThetaData[Run]:
-    """
-    Use :code:`transform_coords` to find the theta values for the events and
-    potentially add ORSO metadata.
-
-    Parameters
-    ----------
-    data_array:
-        Data array to convert.
-    graph:
-        Graph for :code:`transform_coords`.
-
-    Returns
-    -------
-    :
-        New data array with theta coordinate.
-    """
-    data_array_theta = data_array.transform_coords(['theta'], graph=graph)
-    #    # Determine if 'gravity' is in the graph and if to add the gravity correction
-    #    if any(
-    #        [
-    #            'gravity' in i.parameters.keys()
-    #            for i in map(inspect.signature, graph.values())
-    #        ]
-    #    ):
-    #        data_array_theta.attrs['orso'].value.reduction.corrections += [
-    #            'gravity correction'
-    #        ]
-    # except ImportError:
-    #    orso.not_found_warning()
-    return ThetaData[Run](data_array_theta)
-
-
-def theta_to_q(
-    data_array: FootprintCorrectedData[Run],
-    q_bins: QBins,
-    graph: SpecularReflectionCoordTransformGraph[Run],
-) -> QData[Run]:
-    """
-    Convert from theta to Q and if necessary bin in Q.
-
-    Parameters
-    ----------
-    data_array:
-        Data array to convert.
-    q_edges:
-        The lower and upper limits for the Q.
-    graph:
-        Graph for :code:`transform_coords`.
-
-    Returns
-    -------
-    :
-        New data array with theta coordinate.
-    """
-    data_array = data_array.transform_coords(["Q"], graph=graph)
-    data_array = data_array.bin({q_bins.dim: q_bins})
-    return QData[Run](data_array)
-
-
-def histogram(data_array: QData[Run]) -> HistogrammedQData[Run]:
-    """
-    Sum the event bins.
-
-    Parameters
-    ----------
-    data_array:
-        Data array to be summed.
-
-    Returns
-    -------
-    :
-        Summed data array.
-    """
-    return HistogrammedQData[Run](data_array.hist())
+) -> FullData[Run]:
+    da = da.transform_coords(["theta", "wavelength", "Q"], graph=graph)
+    da.coords['z_index'] = da.coords['detector_number'] // sc.scalar(64, unit=None)
+    da.coords['y_index'] = da.coords['detector_number'] % sc.scalar(64, unit=None)
```

Comment:
Hard-coded 64 in generic `reflectometry` module (or any module)? Why do we even need these indices in the first place? Can we `fold` the `detector_number` after loading into Z and Y instead?

Reply:
Yes that should definitely change once we bin the events by detector pixel.

I had some problems working with the binned events, that's why I made it flat just to get something to work.